### PR TITLE
dustin model round1

### PR DESCRIPTION
**Pull request by dustinpierce**
This pull request contains the work for the first round of modeling and addressing the class imbalance problem. Highlights:

1. I engineered dummy variables for each categorical variable except `smoking_status`
2. I removed all `NA` values
3. I filtered on `bmi < 66`
4. I scaled `bmi`, `age` and `avg_glucose_level` using "standard scalar" (a.k.a *Z-scores*)
5. I added a section **Stroke cases** to understand statistical differences between those that have a stroke and those that do not. 
6. I used a package called the [ROSE package](https://cran.r-project.org/web/packages/ROSE/ROSE.pdf) to address class imbalance; I used `p = 0.75` to partition the data into non-stroke and stroke cases -- which utilizes both over sampling and under sampling.
7. For first round of model training, I just applied an 80% - 20% partition into **train** and **test** sets for preliminary models.
8. I trained logistic regression, neural network and random forest binary classification models with a confusion matrix for each. The logistic regression and neural network seem reasonable with accuracy ~ 83%. However, the random forest returned a *perfect* fit, i.e. **100%** which was a concern. I tried re-organizing standardization, feature engineering steps, etc. but could not determine the cause. 

**Next steps**
- Thorough check of completed work thus far
- Determine cause of perfect fit for random forest model
- Automate data import from Kaggle
- Determine where and how to save models
- Execute a robust model training step -- 10-fold cross validation for each model type
- Investigate higher order terms (interaction, squares, etc.) and importance ranking (gini index, PCA)

Let me know if you have questions or comments. Thanks.

### CODE REVIEW

**Comment by caseyag** [comment]
I'm intrigued by the neural networks--looks great Dustin! We can look more into the issues with the Random Forest.
